resdk/resolwe.py

```diff
@@ -241,6 +242,63 @@ def _process_file_field(self, path):
             'file_temp': file_temp,
         }
 
+    def _get_process(self, slug=None):
+        """Return process with given slug.
+
+        Raise error if process doesn't exist or more than one is returned
+        """
+        process = self.api.process.get(slug=slug, ordering='-version', limit=1)
+
+        if len(process) == 1:
+            process = process[0]
+        elif len(process) == 0:
+            raise ValueError("Could not get process for given slug.")
+        else:
+            raise ValueError("Unexpected behaviour at get process with slug {}".format(slug))
+
+        return process
+
+    def _process_inputs(self, inputs, process):
+        """Process input fields."""
+        inputs = copy.deepcopy(inputs)  # leave original intact
+
+        try:
+            for schema, fields in iterate_fields(inputs, process['input_schema']):
+                field_name = schema['name']
+                field_type = schema['type']
+                field_value = fields[field_name]
+
+                # Wrap `list:` fields into list if they are not already
+                if field_type.startswith('list:') and not isinstance(field_value, list):
```

Comment:
We should support this on the server. If you insist on this here, add an XXX comment to remove this when we support it on the server.